doc/website/blog/2021-11-10-announcing-entoas.md

```diff
@@ -0,0 +1,270 @@
+described [here](https://entgo.io/docs/code-gen#use-entc-as-a-package). First install the extension to your Go module:
+
+```shell
+go get entgo.io/contrib/entoas
+```
+
+Now follow the next two steps to enable it and to configure Ent to work with the `elk` extension:
+
+1\. Create a new Go file named `ent/entc.go` and paste the following content:
+
+```go
+// +build ignore
+
+package main
+
+import (
+	"log"
+
+	"entgo.io/contrib/entoas"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+)
+
+func main() {
+	ex, err := entoas.NewExtension()
+	if err != nil {
+		log.Fatalf("creating entoas extension: %v", err)
+	}
+	err = entc.Generate("./schema", &gen.Config{}, entc.Extensions(ex))
+	if err != nil {
+		log.Fatalf("running ent codegen: %v", err)
+	}
+}
+
+```
+
+2\. Edit the `ent/generate.go` file to execute the `ent/entc.go` file:
+
+```go
+package ent
+
+//go:generate go run -mod=mod entc.go
+```
+
+With these steps complete, all is set up for generating an OAS document from your schema! If you are new to Ent and want
+to learn more about it, how to connect to different types of databases, run migrations or work with entities, then head
+over to the [Setup Tutorial](https://entgo.io/docs/tutorial-setup/).
+
+### Generate an OAS document
+
+The first step on our way to the OAS document is to create an Ent schema graph. For the sake of brevity here is an
+example schema to use:
+
+Let's create our schema:
+
+```go title="ent/schema/schema.go"
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+)
+
+// Fridge holds the schema definition for the Fridge entity.
+type Fridge struct {
+	ent.Schema
+}
+
+// Fields of the Fridge.
+func (Fridge) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title"),
+	}
+}
+
+// Edges of the Fridge.
+func (Fridge) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("compartments", Compartment.Type),
+	}
+}
+
+// Compartment holds the schema definition for the Compartment entity.
+type Compartment struct {
+	ent.Schema
+}
+
+// Fields of the Compartment.
+func (Compartment) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Compartment.
+func (Compartment) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("fridge", Fridge.Type).
+			Ref("compartments").
+			Unique(),
+		edge.To("contents", Item.Type),
+	}
+}
+
+// Item holds the schema definition for the Item entity.
+type Item struct {
+	ent.Schema
+}
+
+// Fields of the Item.
+func (Item) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Item.
+func (Item) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("compartment", Compartment.Type).
+			Ref("contents").
+			Unique(),
+	}
+}
+```
+
+Now run the code generator:
+
+```shell
+go generate ./...
+```
+
+In addition to the files Ent normally generates, another file named `ent/openapi.json` has been created. Copy its contents
```

Comment:
Show the output JSON file ! (can be truncated so its small but should demonstrate the features of the extension without needing to run i t).